### ISSUE TITLE
Schema error for var_naming_pattern in Git Repositories when auto-detection mode is used

### ISSUE THREAD

**Issue by ombre8** [ansible/ansible-lint#2751]
##### Summary
[Ansible Lint Docs](https://ansible-lint.readthedocs.io/configuring/) example Config states a `var_naming_pattern` configuration as followed:

```yaml
# Enforce variable names to follow pattern below, in addition to Ansible own
# requirements, like avoiding python identifiers. To disable add `var-naming`
# to skip_list.
# var_naming_pattern: "^[a-z_][a-z0-9_]*$"
```
But activating this config in git repositories results in 

```bash
schema[ansible-lint-config]: Additional properties are not allowed ('var_naming_pattern' was unexpected) (warning)
```

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

This Issue probably exists since the introduction of JSON-Schemas, since I couldn't find `var_naming_pattern` in any Version of [ansible-lint-config.json](https://github.com/ansible/ansible-lint/blob/main/src/ansiblelint/schemas/ansible-lint-config.json)

Or is relatet to: https://github.com/ansible/ansible-lint/issues/763

##### STEPS TO REPRODUCE

Add `var_naming_pattern: "^[a-z_][a-z0-9_]*$"` from the Config-Documentation to your Ansible-Lint config file and run ansible-lint

##### Desired Behavior

It is possible to check patterns of all Variables.

##### Actual Behavior

Test playbook:
Not needed

Configure Ansible-lint

```bash
$ echo 'var_naming_pattern: "^[a-z_][a-z0-9_]*$"' > .ansible-lint
```

Run `ansible-lint` :
```bash
$  ansible-lint -v --force-color
INFO     Set ANSIBLE_LIBRARY=/home/ombre/.cache/ansible-compat/120908/modules:/home/ombre/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/home/ombre/.cache/ansible-compat/120908/collections:/home/ombre/.ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/home/ombre/.cache/ansible-compat/120908/roles:/home/ombre/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles
INFO     Looking up for files, excluding .cache|.git|.hg|.svn|.tox ...

Passed with production profile: 0 failure(s), 0 warning(s) on 1 files.
```
So far so good, now initialize a git repository:

```bash
$ git init 
Initialized empty Git repository in /home/ombre/tmp/ansible-lint-check2/.git/
```

And rerun ansible-lint again:

```bash
$ ansible-lint -v --force-color
INFO     Set ANSIBLE_LIBRARY=/home/ombre/.cache/ansible-compat/120908/modules:/home/ombre/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/home/ombre/.cache/ansible-compat/120908/collections:/home/ombre/.ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/home/ombre/.cache/ansible-compat/120908/roles:/home/ombre/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles
INFO     Discovered files to lint using: git ls-files --cached --others --exclude-standard -z
INFO     Excluded removed files using: git ls-files --deleted -z
WARNING  Listing 1 violation(s) that are fatal
schema[ansible-lint-config]: Additional properties are not allowed ('var_naming_pattern' was unexpected) (warning)
.ansible-lint:1  Returned errors will not include exact line numbers, but they will mention
the schema name being used as a tag, like ``schema[playbook]``,
``schema[tasks]``.

This rule is not skippable and stops further processing of the file.

Schema bugs should be reported towards [schemas](https://github.com/ansible/schemas) project instead of ansible-lint.

If incorrect schema was picked, you might want to either:

* move the file to standard location, so its file is detected correctly.
* use ``kinds:`` option in linter config to help it pick correct file type.


You can skip specific rules or tags by adding them to your configuration file:
# .config/ansible-lint.yml
warn_list:  # or 'skip_list' to silence them completely
  - experimental  # all rules tagged as experimental

                         Rule Violation Summary
 count tag                         profile rule associated tags
     1 schema[ansible-lint-config] basic   core, experimental (warning)

Passed with min profile: 0 failure(s), 1 warning(s) on 2 files.
```

Also, this bug only happens, when there are no `lintables` defined and auto-discovery mode is used when calling `ansible-lint`
